pkg/engine/engine.go

```diff
@@ -1057,15 +1026,9 @@ func (e *Engine) verificationOverlapWorker(ctx context.Context, hasher hasher.Ha
 
 						cacheVal, exists := e.detectionCache.Get(string(keyHash))
 						if exists {
-							// Update the "res" object with the cached result.
-							res.Verified = cacheVal.Verified
-							res.DecoderType = cacheVal.DecoderType
-							res.DetectorType = cacheVal.DetectorType
-							res.ExtraData = cacheVal.ExtraData
-							res.Redacted = cacheVal.Redacted
-							res.StructuredData = cacheVal.StructuredData
-							res.AnalysisInfo = cacheVal.AnalysisInfo
-							// TODO: figure out how to set verification error.
+							// Update the cached result with the Raw and RawV2 values.
+							cacheVal.Raw = res.Raw
+							cacheVal.RawV2 = res.RawV2
```

Comment:
`detectionCache` is storing pointers, right? If that's the case, then I think that means setting values on `cacheVal` will result in them being available from the cache.

Do you think that storing `detectors.Result` in the cache instead of `*detectors.Result` would be a viable solution?

Reply:
I initially considered that, but opted for the pointer to reduce the size from 136B to 8B per value. Do you think the increased memory footprint is okay?

Reply:
Hmm.. I don't know! Do larger values affect the performance? In terms of overall memory usage I would expect it to be roughly the same, since the pointers need to point to actual data.

Reply:
I thought that using a value instead of a pointer would double memory usage by creating a copy of the struct, and that the extra memory would persist until the gc cleans up the unused struct. You are thinking the increased memory in that case is negligible because the original Result will only stay in memory briefly before being gc'd?

Reply:
Ah, I see. Yeah that makes sense that a copy of the data would be stored in the cache. I hadn't thought about the original being heap allocated.